tests/helpers_tests/test_condition.py

```diff
@@ -126,264 +130,192 @@ def test_evaluate__returns_null__when_condition_has_an_invalid_type_property(sel
-
-        self.assertRaises(Exception)
-
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid2(self):
-
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': "+--"}, self.mock_client_logger
-        )
-
-        self.assertRaises(Exception)
-
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid3(self):
-
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': "...+"}, self.mock_client_logger
-        )
-
-        self.assertRaises(Exception)
-
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid4(self):
-
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': "+test"}, self.mock_client_logger
-        )
-
-        self.assertRaises(Exception)
-
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid5(self):
-
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': "3.6"}, self.mock_client_logger
-        )
-
-        self.assertRaises(Exception)
-
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid6(self):
-
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': "2"}, self.mock_client_logger
-        )
-
-        self.assertRaises(Exception)
+    def test_evaluate__returns_true__when_user_version_matches_target_version(self):
+        user_versions = ['2.0.0', '2.0']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_equal_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertEquals(result, True, custom_err_msg)
 
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid7(self):
+    def test_evaluate__returns_false__when_user_version_matches_target_version(self):
+        user_versions = ['2.9', '1.9']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_equal_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertEquals(result, False, custom_err_msg)
 
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': "3.90"}, self.mock_client_logger
-        )
+    def test_evaluate__returns_true__when_user_version_less_than_or_equal_to_target_version(self):
```

Comment:
rename all tests to something that shows the exact evaluator being used. 
test_semver_eq__returns _true
test_semver_gt__returns_false